### DISCUSSION TITLE
repeat leading zero

### DISCUSSION THREAD

**Discussion by pr-apes** [speedata/publisher#452] [Q&A] [answered]
@pgundlach,

inside a `<ForAll>` loop, I want to add leading zeros such as in:

``` xml
<Value select="concat(substring('00000000000000000', 1, string-length(last())-string-length(position())), position(), '-', string(cell_1))"/>
```

Is there no more elegant way to repeat a string a given number of times than abusing `substring`?

Many thanks for your help.

**Comment by pgundlach** [comment, marked as answer]
I don't think you can make that much better.

**Reply by pr-apes** [reply]
I thought my approach was crappy code.

But I see now this is not that bad (as far as it goes).
